autogen/variables.tf.tmpl

```diff
@@ -236,3 +235,52 @@ variable "random_certificate_suffix" {
   type        = bool
   default     = false
 }
+{% if backend_bucket %}
+variable "bucket_name" {
+  description = "a string value representing the name of the cloud storage bucket for the backend"
+  type        = string
+  default     = null
+}
+
+variable "enable_cdn_for_bucket" {
```

Comment:
I think these CDN options are valid for *all* backends (not just buckets), so we shouldn't gate them here. Maybe just remove them for now so we can discuss in a follow-up PR.

Reply:
it seems to be specific to the Backend_Bucket resource based on the resource variables: https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_backend_bucket. I'm open to removing it, but we should discuss first.